test/unit/org/apache/cassandra/sensors/RequestSensorsTest.java

```diff
@@ -140,12 +140,24 @@ public void testSyncAll()
         context1Sensors.registerSensor(context1, type1);
         context1Sensors.registerSensor(context1, type2);
 
-        context1Sensors.getSensor(context1, type1).get().increment(1.0);
-        context1Sensors.getSensor(context1, type2).get().increment(1.0);
+        context1Sensors.incrementSensor(context1, type1, 1.0);
+        context1Sensors.incrementSensor(context1, type2, 1.0);
 
         context1Sensors.syncAllSensors();
-        verify(sensorsRegistry, times(1)).updateSensor(eq(context1), eq(type1), eq(1.0));
-        verify(sensorsRegistry, times(1)).updateSensor(eq(context1), eq(type2), eq(1.0));
+        verify(sensorsRegistry, times(1)).incrementSensor(eq(context1), eq(type1), eq(1.0));
+        verify(sensorsRegistry, times(1)).incrementSensor(eq(context1), eq(type2), eq(1.0));
+
+        // Syncing again doesn't update the sensor
+        context1Sensors.syncAllSensors();
+        verify(sensorsRegistry, times(1)).incrementSensor(eq(context1), eq(type1), eq(1.0));
+        verify(sensorsRegistry, times(1)).incrementSensor(eq(context1), eq(type2), eq(1.0));
```

Comment:
I think these `verify` checks only tell us that the `incrementSensor` metric is not called with a delta of  `1.0`, but IIUC they won't fail if they are called with a different increment, e.g. there's a bug? To be extra safe we might check that the value of the sensor in the registry is actually what we expect after syncing. Wdyt?

Reply:
> they won't fail if they are called with a different increment

Good point.

> To be extra safe we might check that the value of the sensor in the registry is actually what we expect after syncing. 

Well the sensor registry is a mock here, so we can't really do that? That is why I added that new test in `SensorRegistryTest`.

Reply:
Right, then maybe we can add the `requestSensors.syncAllSensors();` without a previous update in that test too and check the registry value didn't change after that?

Reply:
See: https://github.com/datastax/cassandra/pull/1119/commits/9de62fcc76979eef6cee0a58b9f21ce4f6019cfa

Reply:
fwiw, another way to mitigate `they won't fail if they are called with a different increment` is `verifyNoMoreInteractions` on the mock. For me the `SensorRegistryTest` is the real deal.